core/src/main/java/io/methvin/watcher/changeset/ChangeSetListener.java

```diff
@@ -13,7 +18,25 @@
 
   private final Object lock = new Object() {};
 
-  public ChangeSetListener() {}
+  private final int timeout;
+
+  private ScheduledExecutorService service;
+  private ScheduledFuture currentTask;
+
+  private Consumer<Integer> onIdleListener;
+
+  public ChangeSetListener() {
+    this(-1, null);
+  }
+
+  public ChangeSetListener(int timeout, Consumer<Integer> onIdleListener) {
+    this.timeout = timeout;
+    this.onIdleListener = onIdleListener;
+
+    if (timeout > 0) {
```

Comment:
I actually think there's a valid use case for `timeout == 0`, in case you want to be notified as soon as we hit an idle state.

Reply:
yeap, agree